Favor Calculator Implementation

nick-mynatt opened the pull request:
More verbose description in each individual commit

justincpresley commented on ndn_hydra/repo/group_messages/heartbeat.py:
```diff
@@ -13,14 +13,17 @@
     node_name = BytesField(HeartbeatMessageTypes.NODE_NAME)
-    favor = BytesField(HeartbeatMessageTypes.FAVOR)
+    latitude = BytesField(HeartbeatMessageTypes.LATITUDE)
+    longitude = BytesField(HeartbeatMessageTypes.LONGITUDE)
```
First, we need a structure called FavorParamaters that includes lat/long and other information.
We can then include that struct in ALL messages via the ModelType tlv. In addition, we need to update the global view every message.

nick-mynatt replied:
Hi, please check out my most recent commit and let me know what you think. In regards to updating the global view, do you mean that just for the heartbeat messages? Or ALL messages? If you mean just the heartbeat, I believe in the apply function in the Heartbeat message it updates the global view with the new calculated favor.

![Screenshot of 4 nodes running with different coordinates in client](https://user-images.githubusercontent.com/62191506/191871872-df2410c0-8406-4da6-84e2-6ccb6f86ca4b.png)
![The node table in global view of node1](https://user-images.githubusercontent.com/62191506/191871947-51868ff0-f1ba-41f0-a333-85049d273447.png)

nick-mynatt commented:
Two commits since my last update:
1. Introduced FavorParameters class
2. Updated that class with better parameters for calculating favor

Here is an example of two nodes calculating the other's favor
![Screenshot from 2022-09-28 18-09-58](https://user-images.githubusercontent.com/62191506/192904859-506d3450-52bb-4771-8dc7-876fbe3344da.png)

justincpresley commented:
Looks good. Could you run through the examples? You can just state when its good to merge.

nick-mynatt commented:
Ran through examples with no issues. Should be good to merge!

![Screenshot from 2022-09-30 23-14-56](https://user-images.githubusercontent.com/62191506/193392100-fd2bc3bb-a5c2-4934-989d-56b717b6add2.png)
